From 86e0581e5c28de102237aad0182a9ed96801c3a4 Mon Sep 17 00:00:00 2001
From: Jerome Glisse <jglisse@redhat.com>
Date: Thu, 19 Feb 2015 14:48:51 -0500
Subject: [PATCH] leak

---
 src/drmmode_display.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/drmmode_display.c b/src/drmmode_display.c
index 40afb49..cebd206 100644
--- a/src/drmmode_display.c
+++ b/src/drmmode_display.c
@@ -576,6 +576,7 @@ done:
 	else
 		crtc->active = TRUE;
 #endif
+	free(output_ids);
 
 	return ret;
 }
@@ -1374,6 +1375,9 @@ drmmode_output_init(ScrnInfoPtr pScrn, drmmode_ptr drmmode, drmModeResPtr mode_r
 			drmmode_output = output->driver_private;
 			drmmode_output->output_id = mode_res->connectors[num];
 			drmmode_output->mode_output = koutput;
+			for (i = 0; i < koutput->count_encoders; i++)
+				drmModeFreeEncoder(kencoders[i]);
+			free(kencoders);
 			return;
 		}
 	}
-- 
1.8.3.1

