diff --git a/src/drmmode_display.c b/src/drmmode_display.c
index 2258e7a..4571bf0 100644
--- a/src/drmmode_display.c
+++ b/src/drmmode_display.c
@@ -44,7 +44,8 @@
 #include "radeon_reg.h"
 
 #ifdef RADEON_PIXMAP_SHARING
-#include <dri.h>
+#include <dri2.h>
+#include "radeon_dri2.h"
 #endif
 
 #include "drmmode_display.h"
@@ -2409,7 +2410,7 @@ Bool drmmode_pre_init(ScrnInfoPtr pScrn, drmmode_ptr drmmode, int cpp)
 	drmmode_clones_init(pScrn, drmmode, mode_res);
 
 #ifdef RADEON_PIXMAP_SHARING
-	bus_id_string = DRICreatePCIBusID(info->PciInfo);
+	bus_id_string = radeon_dri2_create_pci_bus_id(info->PciInfo);
 	XNFasprintf(&provider_name, "%s @ %s", pScrn->chipset, bus_id_string);
 	free(bus_id_string);
 	xf86ProviderSetup(pScrn, NULL, provider_name);
diff --git a/src/radeon_dri2.c b/src/radeon_dri2.c
index d0dcf89..0996425 100644
--- a/src/radeon_dri2.c
+++ b/src/radeon_dri2.c
@@ -1563,5 +1563,18 @@ void radeon_dri2_close_screen(ScreenPtr pScreen)
     drmFree(info->dri2.device_name);
 }
 
+/* DRI provided a function for this but DRI2 doesn't, so define here */
+char *radeon_dri2_create_pci_bus_id(const struct pci_device *pci_dev)
+{
+    char *bus_id;
+
+    if (asprintf(&bus_id, "pci:%04x:%02x:%02x.%d",
+                 pci_dev->domain, pci_dev->bus, pci_dev->dev,
+                 pci_dev->func) == -1)
+        return NULL;
+
+    return bus_id;
+}
+
 #endif /* DRI2 */
 
diff --git a/src/radeon_dri2.h b/src/radeon_dri2.h
index 9ba47c7..84f972d 100644
--- a/src/radeon_dri2.h
+++ b/src/radeon_dri2.h
@@ -42,11 +42,13 @@ struct radeon_dri2 {
 #include "dri2.h"
 Bool radeon_dri2_screen_init(ScreenPtr pScreen);
 void radeon_dri2_close_screen(ScreenPtr pScreen);
+char *radeon_dri2_create_pci_bus_id(const struct pci_device *pci_dev);
 
 #else
 
 static inline Bool radeon_dri2_screen_init(ScreenPtr pScreen) { return FALSE; }
 static inline void radeon_dri2_close_screen(ScreenPtr pScreen) {}
+static inline char *radeon_dri2_create_pci_bus_id(const struct pci_device *pci_dev) {}
 
 #endif
 
diff --git a/src/radeon_probe.c b/src/radeon_probe.c
index aaace2b..f57dd84 100644
--- a/src/radeon_probe.c
+++ b/src/radeon_probe.c
@@ -47,8 +47,9 @@
 
 #include "xf86.h"
 
+#include "xf86drm.h"
 #include "xf86drmMode.h"
-#include "dri.h"
+#include "dri2.h"
 
 #if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 #include <xf86_OSproc.h>
@@ -64,6 +65,8 @@
 
 #include "radeon_pci_device_match_gen.h"
 
+#include "radeon_dri2.h"
+
 _X_EXPORT int gRADEONEntityIndex = -1;
 
 /* Return the options for supported chipset 'n'; NULL otherwise */
@@ -88,13 +91,7 @@ static Bool radeon_kernel_mode_enabled(ScrnInfoPtr pScrn, struct pci_device *pci
     char *busIdString;
     int ret;
 
-    if (!xf86LoaderCheckSymbol("DRICreatePCIBusID")) {
-      xf86DrvMsgVerb(pScrn->scrnIndex, X_INFO, 0,
-		   "[KMS] No DRICreatePCIBusID symbol, no kernel modesetting.\n");
-	return FALSE;
-    }
-
-    busIdString = DRICreatePCIBusID(pci_dev);
+    busIdString = radeon_dri2_create_pci_bus_id(pci_dev);
     ret = drmCheckModesettingSupported(busIdString);
 #if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
     if (ret) {
